comms/dht/src/event.rs

```diff
@@ -20,13 +20,19 @@
 // WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 // USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+use crate::network_discovery::DhtNetworkDiscoveryRoundInfo;
 use std::sync::Arc;
 use tokio::sync::broadcast;
 
 pub type DhtEventSender = broadcast::Sender<Arc<DhtEvent>>;
 pub type DhtEventReceiver = broadcast::Receiver<Arc<DhtEvent>>;
 
 #[derive(Debug)]
+#[non_exhaustive]
```

Comment:
I'm not sure about this. Is it not safer to force a breaking change when new variants are added?

Reply:
I don't think it makes any kind of difference that matters in our case because this library forms part of one codebase - tari_comms_dht is a library, and we're saying here that client code should handle the case where they don't know a particular event (client code must include a `_ => {}` branch) so that more events can be added later without making a breaking change . Generally, users will do this anyway as their component is only interested in a few events but not all - this just forces a user to handle that case (typically by just ignoring the event). 

Does the use of non_exhaustive here increase the likelihood of bugs being introduced down the line? I'd say it's neutral.